CHANGELOG.md

```diff
@@ -1,7 +1,20 @@
 # Change Log (Release Notes)
 
 All _notable_ changes to this project will be documented in this file (`CHANGELOG.md`).
-This project adheres to [the Python form of Semantic Versioning](https://packaging.python.org/tutorials/distributing-packages/#choosing-a-versioning-scheme) (or at least we try).
+
+This project adheres to [Semantic Versioning 2](http://semver.org/spec/v2.0.0.html), the BigchainDB **public API**:
+* Includes:
+  * This definition of **public API**.
+  * The Command Line Interface.
+  * The transaction schema.
+  * All the endpoints exposed by the HTTP API (including payloads sent to and received from them).
+  * The docker setup.
+* Excludes:
```

Comment:
I think there is even a distinction between `experimental` and `external`.

We might want to never provide backwards compatibility to some of our components, but these could be fairly stable (the opposite of `experimental`).

E.g. As we see transactions and assets as the main interaction paradigm of BigchainDB, we could view blocks and votes as `external`, hence not give backwards compatibility guarantees regarding semver.
We can't really make that change now anymore however, as we've already promised backwards-compatibility to our data models in general: https://github.com/bigchaindb/bigchaindb/pull/1547/files#diff-57fcbb9dfad73b9008ffed4bbbb799ddR10